server/src/events.mjs

```diff
@@ -92,10 +93,79 @@ function sendEventListenerAck(ws, oMsg) {
   logger.debug(`Sent event listener ack message: ${ackMessage}`);
 }
 
+// sendEvent function to handle pre- post- method/event triggers
+function sendEvent(ws, method, result, msg, fSuccess, fErr, fFatalErr) {
+  try {
+    if ( !isRegisteredEventListener(method) ) {
+      fErr.call(null, method);
+
+    } else {
+       // Fire pre trigger if there is one for this method
+       if ( method in eventTriggers ) {
+        if ( 'pre' in eventTriggers[method] ) {
+          try {
+            const ctx = {
+              logger: logger,
+              setTimeout: setTimeout,
+              setInterval: setInterval
+            };
```

Comment:
We should pass in a 'sendEvent' method in the ctx object so that a developer can use it send a different event (on top of the one for which the trigger is firing).

Reply:
added sendEvent in the ctx object